stock_inventory_discrepancy/hooks.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2019 Eficent Business and IT Consulting Services S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+
+from odoo import _
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare
+
+from odoo.addons.stock.models.stock_inventory import Inventory
+
+
+def post_load_hook():
+    def action_validate_discrepancy(self):
+        """Override method to avoid inline group validation"""
+        if not self.exists():
+            return
+        self.ensure_one()
+        # START HOOK: - Allow specific group to validate inventory
+        #             - Allow validate on pending status
+        if (
+            not self.user_has_groups("stock.group_stock_manager")
+            and not self.user_has_groups(
+                "stock_inventory_discrepancy.group_stock_inventory_validation"
+            )
+            and not self.user_has_groups(
+                "stock_inventory_discrepancy.group_stock_inventory_validation_always"
+            )
+        ):
+            raise UserError(
+                _("Only a stock manager can validate an inventory adjustment.")
+            )
+        if self.state not in ["confirm", "pending"]:
```

Comment:
In which case do you need the inventory pending at validation (As I understand, the only thing you change with this hook ) ?

Why not having something that make pending inventory automatically confirmed ?

A little bit too much code copy for that.

Reply:
It is required for the module to run properly